lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -98,8 +116,10 @@ def execute(self):
                             else ""
                         )
                         found = True
+
             if found:
-                self.print_results(tool_info, tool, self.context.with_option)
+                found = self.print_results(tool_info, self.context.with_option)
+                assert found, "Tool found but not printed!"
```

Comment:
I don't see how this `assert` (or being able to write it) actually helps anyone. We didn't print anything and we're about to exit after finding something we think should have been printed; we abort with "found is False because we found something but didn't print it" ... it doesn't strike me as being useful diagnostic context because it only re-states the obvious in what seems an unnecessarily fatal manner for a transient CLI command.

Reply:
One of the things I was trying to accomplish with these changes is to remove the cases where the command completes silently.  I'm of the opinion (which might, of course, be _wrong_...) that this command should either issue the requested information on `stdout` or a diagnostic on `stderr` -- it shouldn't complete without a comment, one way or the other.  (I.e., either the user gets what they asked for, or they get told why they were wrong to ask!  😀)

Eventually, I hit that spot in the code, where it _should_ have printed something, and `found` _shouldn't_ ever be false, there, so I figured I would put that assertion in for testing purposes.

I'll take it out (along with the reassignment of `found`) before I open this PR for review.

Reply:
It's gone now.  (I never saw it trigger during testing, as I'd hoped would be the case....)